### PR TITLE
Add recent scores section to user profile

### PR DESCRIPTION
**Pull request by jorolf**

### CODE REVIEW

File: osu.Game/Overlays/Profile/Sections/Ranks/DrawableScore.cs [hunk 1]
```diff
@@ -159,5 +135,61 @@ protected override IEnumerable<Vector2> ComputeLayoutPositions()
                     yield return new Vector2(DrawWidth * i * (count == 1 ? 0 : 1f / (count - 1)), 0);
             }
         }
+
+        public class PPScore : DrawableScore
```
Comment (peppy) [inline osu.Game/Overlays/Profile/Sections/Ranks/DrawableScore.cs]: let's move these two to their own classes

File: osu.Game/Overlays/Profile/Sections/Ranks/ScoreContainer.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                    }
+                }
+            };
+
+            Schedule(() => { api.Queue(req); });
+        }
+
+        protected abstract DrawableScore CreateScore(OnlineScore score, int index);
+
+        public class PPScoreContainer : ScoreContainer
```
Comment (peppy) [inline osu.Game/Overlays/Profile/Sections/Ranks/ScoreContainer.cs]: what's with the `ScoreContainer` thing? Do these need to be typed?

File: osu.Game/Overlays/Profile/Sections/HistoricalSection.cs [hunk 3]
```diff
@@ -1,12 +1,29 @@
...
         public override string Identifier => "historical";
+
+        private readonly ScoreContainer recent;
+
+        public HistoricalSection()
+        {
+            Child = recent = new ScoreContainer.TotalScoreContainer(ScoreType.Recent, "Recent Plays (24h)", "No performance records. :(");
+        }
+
+        public override User User
```
Comment (peppy) [inline osu.Game/Overlays/Profile/Sections/HistoricalSection.cs]: I was thinking this with the last PR, but we probably want to use a bindable pattern for `User` here so we don't have to do these increasingly large chains of `set`s. Let me know if you're unsure of how to implement this.

File: osu.Game/Overlays/Profile/Sections/Ranks/DrawableScore.cs (hunk 1 above)
Comment (jorolf) [reply osu.Game/Overlays/Profile/Sections/Ranks/DrawableScore.cs]: I intentionally placed them there so I could make the constructor private. That also avoided the problem of having to set other fields not private

**Review by peppy — APPROVED** [review]

**Comment by peppy** [comment]
@jorolf please look over my changes to make sure you know what needs to be done next time you submit stuff. i made quite a few name changes as well as logic changes, separated into individual commits for your digestion.

**Pull request merged**